### DISCUSSION TITLE
Pointer events in Control vs. Panel.

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#9794] [General]
I want to capture the mouse movement in a custom control for dragging, etc., so I derive from `Control` since no Children.
When I don't draw anything in `Render`, there are no mouse movements captured.
The only way I get mouse events is when the mouse goes over something that has been drawn in `Render`, such as individual lines (I get quick little `PointerEnter` and `PointerLeave` events when mouse crosses a line) or shapes, and I don't get any click events unless I'm over a drawn pixel somewhere.  

I can draw a large rectangle to capture the mouse, or alternately derive from Panel and define a (possibly transparent) Background in .xaml, which seems to have the same effect as me just drawing a rectangle.  

Is this expected?  Is there any way to get mouse events from a completely blank Control? 

My current solution is to derive from Panel (even though I [currently] don't have any Children here), and then specify a blank background.

    public partial class GraphGrid : Control /* or Panel */
    {
        // Control does not have Background Property 
        // Panel does have Background Property

        public GraphGrid()
        {
            InitializeComponent();
            this.PointerMoved += (o, e) => Console.WriteLine("Moved");
            this.PointerEnter += (o, e) => Console.WriteLine("Enter");
            this.PointerLeave += (o, e) => Console.WriteLine("Leave");
            this.PointerPressed += (o, e) => Console.WriteLine("Pressed");
            this.PointerReleased += (o, e) => Console.WriteLine("Release");
	    }

        public override void Render(DrawingContext context)
        {
            // Avalonia draws background for me if deriving from Panel
            // and if Background specified in xaml.

            // No Background property to draw if deriving from Control

            // I only get mouse events over stuff that has been drawn,
            // ie a Background or custom graphics.
            // I don't get mouse events on a completely clear/blank Control
            // or Panel.
        
            base.Render(context);

            // Draw custom graphics
            context.DrawBlaBla(...);

        }
    }

**Comment by timunie** [comment]
You need any kind of Background for hit testing to work. Transparent is fine, but null will not work. For Panel, null is default.

**Comment by sonicsmooth** [comment]
Thanks, I figured this is by design.  I'm just using a Panel with defined background.
